src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
+    }
+
+    render () {
+      return <Target
+        {...this.props}
+        onStartFPSCollection={handleStartFPSCollection}
```

Comment:
Then they have to use `let`, since it will have to be defined outside the component code.

Something like:

```
let endCollection

class AnimatedComponent extends Component {
  componentDidMount () {
    endCollection = this.props.onCollectFPS()
    setTimeout(() => endCollection())
  }
  ...
}
```

isn’t that awful?

In my mind that would also make this hoc redundant. At that point you can really just use `collect-fps` manually.